### PR TITLE
feat(switcher): add component Al 1713

### PR DESCRIPTION
**Pull request by ianaa**
no extra functionality over the headless ui's radio group. Just styling plus icon support.

Please let me know how you like this new pattern for storybook docs. Not sold on it yet, but I like the more structured way of describing the stories which I imagine would result in more consistent docs between components.

### CODE REVIEW

File: src/components/Switcher.vue [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      class="focus-visible:focus-outline flex h-8 rounded transition-colors duration-500 ease-out">
+      <div
+        class="flex items-center px-3 py-1 rounded enabled:active:bg-zurich48 enabled:active:text-mediumblue enabled:cursor-pointer"
+        :class="{
+          'text-mediumblue bg-zurich': checked,
+          'hover:bg-gray': !checked && !option.disabled && !disabled,
+          'opacity-40 cursor-not-allowed': option.disabled || disabled
+        }"
+        :aria-label="option.icon ? option.label : undefined">
+        <slot :name="option.value">
```
Comment (bnjm) [inline src/components/Switcher.vue:83]: Using the value as the slot name is clever! Are there any caveats with dynamic options/slots? We have at least one place where options would be added dynamically to the switcher:

Screenshot 2023-11-14 at 16 41 57

Screenshot 2023-11-14 at 16 42 02

... this case wouldn't even need slots, but still worth making note of it in the docs if it's not possible to dynamically add them

**Review by manuatarchilogic — APPROVED** [review]

File: src/components/Switcher.vue (hunk 1 above)
Comment (ianaa) [reply src/components/Switcher.vue:83]: Good idea!
there are some constraints mentioned in the vue docs, I'll add these links https://vuejs.org/guide/components/slots.html#dynamic-slot-names
https://vuejs.org/guide/essentials/template-syntax.html#directives

**Comment by manuatarchilogic** [comment]
Anything noteworthy to make tooltips work for this new component? (e.g. one might wanna add a tooltip for an icon-only switcher)

**Comment by ianaa** [comment]
> Anything noteworthy to make tooltips work for this new component? (e.g. one might wanna add a tooltip for an icon-only switcher)

@manuatarchilogic excellent question and completely overlooked on my side. you'd have to use the slots to add tooltips as a consumer. 
do you think if I add a title attribute to them it would do for now? Until @michalarchilogic is back to discuss this.

**Comment by michalarchilogic** [comment]
Adding a slot for tooltips is a great idea. We have designs for this but
only using the new tooltips that still wait for Iana’s love.

W dniu śr., 15.11.2023 o 09:21 Iana Andrychowicz ***@***.***>
napisał(a):

**Comment by ianaa** [comment]
ok, thanks! I'll do the title attribute now, and then go work on the tooltip component :D

**Review by bnjm — APPROVED** [review]

**Comment by romiekos** [comment]
nit: I can click on the disabled button and change a background color:
image

but `<a-button>` does not have this power 🙂

**Review by romiekos — APPROVED** [review]

**Pull request merged**